mypy failure with np 2.2.4

Running mypy with the latest numpy (v 2.2.4, released 3 days ago) results in a new failure: 

```python
yt/__init__.py:98: note: In module imported here:
yt/testing.py: note: In function "integrate_kernel":
yt/testing.py:148: error: Argument 1 has incompatible type "ndarray[tuple[int, ...], dtype[Any]]"; expected "float"  [arg-type]
yt/utilities/performance_counters.py: note: In member "print_stats" of class "PerformanceCounters":
yt/utilities/performance_counters.py:73: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
Found 1 error in 1 file (checked 516 source files)
```

--- a/yt/testing.py
+++ b/yt/testing.py
@@ -17,6 +17,7 @@ import matplotlib
 import numpy as np
 from more_itertools import always_iterable
 from numpy.random import RandomState
+from numpy.typing import NDArray
 from unyt.exceptions import UnitOperationError
 
 from yt._maintenance.deprecation import issue_deprecation_warning
@@ -91,7 +92,7 @@ def assert_rel_equal(a1, a2, decimals, err_msg="", verbose=True):
 
 # tested: volume integral is 1.
 def cubicspline_python(
-    x: float | np.ndarray,
+    x: float | NDArray[np.floating],
 ) -> np.ndarray:
     """
     cubic spline SPH kernel function for testing against more
@@ -118,7 +119,9 @@ def cubicspline_python(
 
 
 def integrate_kernel(
-    kernelfunc: Callable[[float], float], b: float, hsml: float
+    kernelfunc: Callable[[float | NDArray[np.floating]], float | NDArray[np.floating]],
+    b: float,
+    hsml: float,
 ) -> float:
     """
     integrates a kernel function over a line passing entirely
@@ -154,11 +157,11 @@ _zeroperiods = np.array([0.0, 0.0, 0.0])
 
 
 def distancematrix(
-    pos3_i0: np.ndarray,
-    pos3_i1: np.ndarray,
+    pos3_i0: NDArray[np.floating],
+    pos3_i1: NDArray[np.floating],
     periodic: tuple[bool, bool, bool] = (True,) * 3,
-    periods: np.ndarray = _zeroperiods,
-) -> np.ndarray:
+    periods: NDArray[np.floating] = _zeroperiods,
+) -> NDArray[np.floating]:
     """
     Calculates the distances between two arrays of points.
 

--- a/yt/testing.py
+++ b/yt/testing.py
@@ -10,14 +10,14 @@ from collections.abc import Callable, Mapping
 from functools import wraps
 from importlib.util import find_spec
 from shutil import which
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, TypeVar
 from unittest import SkipTest
 
 import matplotlib
 import numpy as np
+import numpy.typing as npt
 from more_itertools import always_iterable
 from numpy.random import RandomState
-from numpy.typing import NDArray
 from unyt.exceptions import UnitOperationError
 
 from yt._maintenance.deprecation import issue_deprecation_warning
@@ -92,8 +92,8 @@ def assert_rel_equal(a1, a2, decimals, err_msg="", verbose=True):
 
 # tested: volume integral is 1.
 def cubicspline_python(
-    x: float | NDArray[np.floating],
-) -> np.ndarray:
+    x: float | npt.NDArray[np.floating],
+) -> npt.NDArray[np.floating]:
     """
     cubic spline SPH kernel function for testing against more
     effiecient cython methods
@@ -119,10 +119,12 @@ def cubicspline_python(
 
 
 def integrate_kernel(
-    kernelfunc: Callable[[float | NDArray[np.floating]], float | NDArray[np.floating]],
-    b: float,
-    hsml: float,
-) -> float:
+    kernelfunc: Callable[
+        [float | npt.NDArray[np.floating]], float | npt.NDArray[np.floating]
+    ],
+    b: float | npt.NDArray[np.floating],
+    hsml: float | npt.NDArray[np.floating],
+) -> float | npt.NDArray[np.floating]:
     """
     integrates a kernel function over a line passing entirely
     through it
@@ -150,18 +152,24 @@ def integrate_kernel(
     dx = np.diff(xe, axis=0)
     spv = kernelfunc(np.sqrt(xc**2 + x**2))
     integral = np.sum(spv * dx, axis=0)
-    return pre * integral
+    result = pre * integral
+    if isinstance(result, np.floating):
+        return result.item()
+    return result
 
 
 _zeroperiods = np.array([0.0, 0.0, 0.0])
 
 
+_FloatingT = TypeVar("_FloatingT", bound=np.floating)
+
+
 def distancematrix(
-    pos3_i0: NDArray[np.floating],
-    pos3_i1: NDArray[np.floating],
+    pos3_i0: npt.NDArray[_FloatingT],
+    pos3_i1: npt.NDArray[_FloatingT],
     periodic: tuple[bool, bool, bool] = (True,) * 3,
-    periods: NDArray[np.floating] = _zeroperiods,
-) -> NDArray[np.floating]:
+    periods: npt.NDArray[_FloatingT] = _zeroperiods,
+) -> npt.NDArray[_FloatingT]:
     """
     Calculates the distances between two arrays of points.
 

--- a/yt/testing.py
+++ b/yt/testing.py
@@ -124,7 +124,7 @@ def integrate_kernel(
     ],
     b: float | npt.NDArray[np.floating],
     hsml: float | npt.NDArray[np.floating],
-) -> float | npt.NDArray[np.floating]:
+) -> npt.NDArray[np.floating]:
     """
     integrates a kernel function over a line passing entirely
     through it
@@ -152,10 +152,7 @@ def integrate_kernel(
     dx = np.diff(xe, axis=0)
     spv = kernelfunc(np.sqrt(xc**2 + x**2))
     integral = np.sum(spv * dx, axis=0)
-    result = pre * integral
-    if isinstance(result, np.floating):
-        return result.item()
-    return result
+    return np.atleast_1d(pre * integral)
 
 
 _zeroperiods = np.array([0.0, 0.0, 0.0])

